File: GARI_Umbria_2026-08-03/DotNet/AgronicaCoreStd/AgronicaCoreModelsSTD/anagrafiche/UtenteColdiretti.cs
using System;

namespace AgronicaCoreModelsSTD.anagrafiche
{
    public class UtenteColdiretti
    {
        public string COD_ANAGEN { get; set; }
        public string COD_TIPO_ANAG { get; set; }
        public DateTime? DAT_INIZIO { get; set; }
        public DateTime? DAT_FINE { get; set; }
        public Blocco_Data data { get; set; }

        private int isPersonaGiuridica = 0;
        public int IsPersonaGiuridica
        {
            get
            {
                if (isPersonaGiuridica == 0)
                {
                    if (COD_TIPO_ANAG == "G" && data.SGL_CODFIS.Length != 16)
                        isPersonaGiuridica = 1; // Persona Giuridica
                    else
                        isPersonaGiuridica = 2; // Prsona Fisica
                }
                return isPersonaGiuridica;
            }
        }

        public string GetUserCode()
        {
            if (IsPersonaGiuridica == 2) // è persona fisica
                return data.titolare.COD_ANAGEN;
            else
                return data.COD_ANAGEN;
        }

        public string Nome()
        {
            string result = "";
            if (IsPersonaGiuridica == 2) // è persona fisica
                result = data.titolare.DES_NOME;
            return result;
        }

        public string Cognome()
        {
            string result = "";
            if (IsPersonaGiuridica == 2) // è persona fisica
                result = data.titolare.DES_COGNOME;
            return result;
        }

        public string RagioneSociale()
        {
            string result = "";
            if (IsPersonaGiuridica == 1) // è persona giuridica
                result = data.DES_RAGIONE_SOCIALE;
            return result;
        }

    }

    public class Blocco_Data
    {
        public string COD_ANAGEN { get; set; }
        public string DES_RAGIONE_SOCIALE { get; set; }
        public string SGL_CODFIS { get; set; }
        public string SGL_PIVA { get; set; }
        public Blocco_Titolare titolare { get; set; }
    }

    public class Blocco_Titolare
    {
        public string COD_ANAGEN { get; set; }
        public string DES_NOME { get; set; }
        public string DES_COGNOME { get; set; }
        public string SGL_CODFIS { get; set; }
        public DateTime? DAT_NASCITA { get; set; }
        public string SGL_COMUNE_NASCITA { get; set; }
    }
}

File: GARI_Umbria_2026-08-03/DotNet/AgronicaWebService2010/AgronicaWebApiProfilatore/Model/UtenteColdiretti.vb
Public Class UtenteColdiretti
    Public Property COD_ANAGEN As String
    Public Property COD_TIPO_ANAG As String
    Public Property DAT_INIZIO As DateTime?
    Public Property DAT_FINE As DateTime?
    Public Property data As Blocco_Data

    Public Function GetUserCode() As String
        If COD_TIPO_ANAG = "G" And data.SGL_CODFIS.Length = 16 Then
            Return data.titolare.COD_ANAGEN
        Else
            Return data.COD_ANAGEN
        End If
    End Function
End Class

Public Class Blocco_Data
    Public Property COD_ANAGEN As String
    Public Property DES_RAGIONE_SOCIALE As String
    Public Property SGL_CODFIS As String
    Public Property SGL_PIVA As String
    Public Property titolare As Blocco_Titolare
End Class

Public Class Blocco_Titolare
    Public Property COD_ANAGEN As String
    Public Property DES_NOME As String
    Public Property DES_COGNOME As String
    Public Property SGL_CODFIS As String
    Public Property DAT_NASCITA As DateTime?
    Public Property SGL_COMUNE_NASCITA As String
End Class

